### PR TITLE
Tag raster cache surface and shader as sRGB

### PR DESCRIPTION
**Pull request by mattsarett**
https://github.com/flutter/flutter/issues/7836

### CODE REVIEW

**Comment by mattsarett** [comment]
@brianosman

So I haven't forgotten about this...  it's just that this change is trickier than I thought.  In particular, tagging the raster cache surface as sRGB is changing the behavior of anything that is drawn to that surface (which is everything if we're using the raster cache!).

https://github.com/flutter/engine/pull/3395 actually helps quite a bit.  It means that decoded images will be tagged with a color space - so they will draw "correctly" to the tagged surface (drawing untagged images to a tagged surfaces is "undefined" in Skia - though in practice right now we'll just draw too light).

There are some shadows etc. that look a little different now that the blending is different - there will probably be some tweaks that the UI designers want to make.

The big difference is text: let me take a moment to complain about translucent text...  There are material design specs in existence that instruct designers to draw semi-translucent black text when they really just want gray text.  This is a bit silly - it just means that we are blending black onto white when we draw the text (rather than just copying in opaque gray).  Another effect of semi-translucent black text is that it depends on blend behavior - so a result of changing to color correct blending is that text looks much lighter.

It's expected that we'll see these kinds of issues when changing the behavior of how we draw.  That's why it's so important that we try to get this right as early as possible.  The problem now is trying to come up with a sane process for landing these types of changes.

If we ignore the text issue, the other subtle differences are not so noticeable, except that they continue to pop in and out depending on whether the raster cache is in use... which I think is not ok.

So maybe we can try something like this?
(1) Track down where text color is being specified (I'm not sure if there are global defaults or if this is an app by app decision).  Change the text from translucent black to gray.  Should be no behavior change.
(2) Tag all surfaces as sRGB.  I'm worried we have to get them all at once or there will be strange inconsistencies.
Note: It looks like most sources that need color space tags (meaning they are not implicitly sRGB) are already tagged.  If we miss a couple it is easy to follow up later.
(3) App tweaks to make everything look as intended with the new behavior.

TLDR: Sorry for the essay.  Trying to think about how we can incrementally land color space support in Flutter as smoothly as possible.

**Comment by chinmaygarde** [comment]
Can we close this PR for now?

**Comment by mattsarett** [comment]
Yes of course, sorry.
